commercetools-convenience/src/main/java/io/sphere/sdk/sequencegenerators/CustomObjectBigIntegerNumberGenerator.java

```diff
@@ -0,0 +1,92 @@
+package io.sphere.sdk.sequencegenerators;
+
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.customobjects.CustomObject;
+import io.sphere.sdk.customobjects.CustomObjectDraft;
+import io.sphere.sdk.customobjects.commands.CustomObjectUpsertCommand;
+import io.sphere.sdk.customobjects.queries.CustomObjectByKeyGet;
+import io.sphere.sdk.models.Base;
+import io.sphere.sdk.utils.CompletableFutureUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.math.BigInteger;
+import java.util.concurrent.CompletionStage;
+
+/**
+ Creates an incremental sequence of BigInteger numbers, storing the last used number in a CustomObject. It is thread safe, by using optimistic concurrency control based on the version of the CustomObject.
+ When there is a concurrency exception, it is automatically retried to generate the number (with a maximum number of retries configured in {@link CustomObjectBigIntegerNumberGeneratorConfig}).
+
+ <h3 id="create-type">Create number sequence starting with 1</h3>
+
+ <p>Execution example:</p>
+ {@include.example io.sphere.sdk.sequencegenerators.BigIntegerNumberGeneratorIntegrationTest#firstNumberIsOne()}
+
+ <h3 id="create-type">Create number sequence with a given initial value</h3>
+
+ <p>It is possible to assign an initial value for the sequence</p>
+
+ <p>Execution example:</p>
+ {@include.example io.sphere.sdk.sequencegenerators.BigIntegerNumberGeneratorIntegrationTest#firstNumberCanBeGiven()}
+
+ <h3 id="create-object-with-type">Create a sequence assigning the container and key for the custom object</h3>
+
+ <p>Execution example:</p>
+ {@include.example io.sphere.sdk.sequencegenerators.BigIntegerNumberGeneratorIntegrationTest#customObjectContainerAndKeyCanBeGiven()}
+
+ */
+
```

Comment:
to be picky: I don't think it is common style to have an empty line between the javadoc and the class.